src/lib.rs

```diff
@@ -160,7 +166,46 @@ pub fn open(url: &str) -> Result<Output> {
 /// }
 /// ```
 pub fn open_browser(browser: Browser, url: &str) -> Result<Output> {
-    open_browser_internal(browser, url).and_then(|status| {
+    open_browser_with_options(BrowserOptions {
+        browser: Some(browser), url: url.into(), suppress_output: Some(false)
+    })
+}
+
+impl BrowserOptions {
+    pub fn create(url: &str) -> BrowserOptions {
+        BrowserOptions {
+            browser: None,
+            suppress_output: None,
+            url: url.into()
+        }
+    }
+
+    pub fn create_with_suppressed_output(url: &str) -> BrowserOptions {
+        BrowserOptions {
+            browser: None,
+            suppress_output: Some(true),
+            url: url.into()
+        }
+    }
+}
+
+/// Opens the specified URL on the specific browser (if available) requested. Return semantics are
+/// the same as for [open](fn.open.html).
+///
+/// # Examples
+/// ```no_run
+/// use webbrowser::{open_browser_with_options, BrowserOptions};
+///
+/// if open_browser_with_options(BrowserOptions::create("http://github.com")).is_ok() {
+///     // ...
+/// }
+/// ```
+pub fn open_browser_with_options(options: BrowserOptions) -> Result<Output> {
```

Comment:
Best to expose out a `pub fn open_browser_with_options(browser: Browser, url: &str, options: &BrowserOptions)` - this will keep the role of `BrowserOptions` as a behaviour modifier, while keeping browser and url as explicit parameters